gobblin-temporal/src/main/java/org/apache/gobblin/temporal/cluster/GobblinTemporalTaskRunner.java

```diff
@@ -0,0 +1,411 @@
+      logger.warn("Cannot find host name for Helix instance: {}");
+    }
+
+    TaskRunnerSuiteBase.Builder builder = GobblinConstructorUtils.<TaskRunnerSuiteBase.Builder>invokeLongestConstructor(
+        new ClassAliasResolver(TaskRunnerSuiteBase.Builder.class)
+            .resolveClass(builderStr), this.clusterConfig);
+
+    return builder.setAppWorkPath(this.appWorkPath)
+        .setContainerMetrics(this.containerMetrics)
+        .setFileSystem(this.fs)
+        .setApplicationId(applicationId)
+        .setApplicationName(applicationName)
+        .setContainerId(taskRunnerId)
+        .setHostName(hostName);
+  }
+
+  private Path initAppWorkDir(Config config, Optional<Path> appWorkDirOptional) {
+    return appWorkDirOptional.isPresent() ? appWorkDirOptional.get() : GobblinClusterUtils
+        .getAppWorkDirPathFromConfig(config, this.fs, this.applicationName, this.applicationId);
+  }
+
+  private Config saveConfigToFile(Config config)
+      throws IOException {
+    Config newConf = config
+        .withValue(CLUSTER_APP_WORK_DIR, ConfigValueFactory.fromAnyRef(this.appWorkPath.toString()));
+    ConfigUtils configUtils = new ConfigUtils(new FileUtils());
+    configUtils.saveConfigToFile(newConf, CLUSTER_CONF_PATH);
+    return newConf;
+  }
+
+  /**
+   * Start this {@link GobblinTemporalTaskRunner} instance.
+   */
+  public void start()
+      throws ContainerHealthCheckException {
+    logger.info("Calling start method in GobblinTemporalTaskRunner");
+    logger.info(String.format("Starting in container %s", this.taskRunnerId));
+
+    // Start metric reporting
+    initMetricReporter();
+
+    // Add a shutdown hook so the task scheduler gets properly shutdown
+    addShutdownHook();
+
+    try {
+      for (int i = 0; i < this.numTemporalWorkers; i++) {
+        initiateWorker();
+      }
+    }catch (Exception e) {
+      logger.info(e + " for initiate workers");
+      throw new RuntimeException(e);
+    }
+  }
+
+  private void initiateWorker() throws Exception{
+    logger.info("Starting Temporal Worker");
+
+    String connectionUri = clusterConfig.getString(GobblinTemporalConfigurationKeys.TEMPORAL_CONNECTION_STRING);
+    WorkflowServiceStubs service = TemporalWorkflowClientFactory.createServiceInstance(connectionUri);
+
+    String namespace = ConfigUtils.getString(clusterConfig, GobblinTemporalConfigurationKeys.GOBBLIN_TEMPORAL_NAMESPACE,
+            GobblinTemporalConfigurationKeys.DEFAULT_GOBBLIN_TEMPORAL_NAMESPACE);
+    WorkflowClient client = TemporalWorkflowClientFactory.createClientInstance(service, namespace);
+
+    String workerClassName = ConfigUtils.getString(clusterConfig,
+        GobblinTemporalConfigurationKeys.WORKER_CLASS, GobblinTemporalConfigurationKeys.DEFAULT_WORKER_CLASS);
+    AbstractTemporalWorker worker = GobblinConstructorUtils.invokeLongestConstructor(
+        (Class<AbstractTemporalWorker>) Class.forName(workerClassName), clusterConfig, client);
+    worker.start();
```

Comment:
not sure how important this may for proper resource management, but these workers do have a corresponding `shutdown()` method, I don't see called anywhere.

if challenging to stitch in now, at least leave a TODO